packages/docusaurus/src/server/i18n.ts

```diff
@@ -15,7 +15,10 @@ function getDefaultLocaleLabel(locale: string) {
   // Intl.DisplayNames is ES2021 - Node14+
   // https://v8.dev/features/intl-displaynames
   if (typeof Intl.DisplayNames !== 'undefined') {
-    return new Intl.DisplayNames([locale], {type: 'language'}).of(locale);
+    const languageName = new Intl.DisplayNames(locale, {type: 'language'}).of(
+      locale,
+    );
+    return languageName.charAt(0).toUpperCase() + languageName.slice(1);
```

Comment:
i18n rules are complex 🤪 , not even sure all locale first letters should always be capitalized. But it's not a big deal IMHO as user can provide its own string, we just need a good enough default 👍 

What about replacing `toUpperCase()` by `toLocaleUpperCase()`? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleUpperCase

Reply:
Agreed, just done it.